pulsar-broker/src/main/java/org/apache/pulsar/broker/service/BrokerService.java

```diff
@@ -305,6 +307,12 @@ public BrokerService(PulsarService pulsar, EventLoopGroup eventLoopGroup) throws
                 .newSingleThreadScheduledExecutor(new DefaultThreadFactory("pulsar-stats-updater"));
         this.authorizationService = new AuthorizationService(
                 pulsar.getConfiguration(), pulsar().getPulsarResources());
+        if (pulsar.getConfiguration().isTransactionCoordinatorEnabled()) {
```

Comment:
We have added a transaction thread pool `transactionReplayExecutor`, it's better to consider using one thread pool for transaction buffer and pending ack state.

For example:

Using a unified transaction thread pool.
One topic pinned one thread to handle the transaction buffer operations and pending ack operations.